modules/programs/neomutt.nix

```diff
@@ -133,9 +133,24 @@ let
     '';
 
   registerAccount = account:
-    with account; ''
+    let
+      mailboxes = if account.neomutt.mailboxName == null then
+        "mailboxes"
+      else
+        ''named-mailboxes "${account.neomutt.mailboxName}"'';
+      extraMailboxes = concatMapStringsSep "\n" (extra:
+        if isString extra then
+          ''mailboxes "${account.maildir.absPath}/${extra}"''
+        else if extra.name == null then
+          ''mailboxes "${account.maildir.absPath}/${extra.mailbox}"''
+        else
+          ''
+            named-mailboxes "${extra.name}" "${account.maildir.absPath}/${extra.mailbox}"'')
+        account.neomutt.extraMailboxes;
+    in with account; ''
       # register account ${name}
-      mailboxes "${account.maildir.absPath}/${folders.inbox}"
+      ${mailboxes} "${account.maildir.absPath}/${folders.inbox}"
+      ${extraMailboxes}
```

Comment:
The tests need to be updated to account for this extra newline that always gets added. Also, this entire change needs a new test.

Reply:
Done